src/test/java/org/mskcc/smile/ValidRequestCheckerTest.java

```diff
@@ -335,4 +335,29 @@ public void testRequestSamplesMissingPatientIds() throws Exception {
                 Object[].class);
         Assert.assertTrue(failedSamplesList.length == 3);
     }
+
+    /**
+     * Test for handling request with 2 out of 4 samples having igoComplete set to false.
+     *
+     */
+    @Test
+    public void testRequestJsonWithSamplesHavingFalseIgoComplete() throws Exception {
+        MockJsonTestData requestJson = mockedRequestJsonDataMap
+                .get("mockRequest8FalseIgoComplete");
+
+        String modifiedRequestJson = validRequestChecker
+                .getFilteredValidRequestJson(requestJson.getJsonString());
+        Assert.assertNotNull(modifiedRequestJson);
+
+        Map<String, Object> requestJsonMap = mapper.readValue(modifiedRequestJson, Map.class);
+        Object[] sampleList = mapper.convertValue(requestJsonMap.get("samples"),
+                Object[].class);
+        for (Object sample: sampleList) {
+            Map<String, String> sampleMap = mapper.convertValue(sample, Map.class);
+            Map<String, Object> sampleStatus = mapper.convertValue(sampleMap.get("status"), Map.class);
+            if (sampleMap.get("igoComplete") == Boolean.FALSE.toString()) {
```

Comment:
String comparisons in Java should be done with `.equals(..)` instead of `==` for future reference

```
if (sampleMap.get("igoComplete").equals(Boolean.FALSE.toString()))
```

or something like this could also work and avoids Java string comparisons

```
if (!Boolean.valueOf(sampleMap.get("igoComplete")))
```

Reply:
hm, mine seems to work as this

```
            if (!Boolean.valueOf(sampleMap.get("igoComplete"))) {
                Assert.assertFalse((Boolean) sampleStatus.get("validationStatus"));
            }
```

if you run `mvn clean install` does it still build?

Reply:
If your test passed with no issue, then maybe something is up with my environment setup. So far, I've tried switching Java versions but it didn't help. Let me investigate some more. 

Running `mvn clean install` failed due to failing 1 test, which is this one. Same error: `java.lang.ClassCastException: java.lang.Boolean cannot be cast to java.lang.String`